kernel/data-pipeline/cdc/core/src/main/java/org/apache/shardingsphere/data/pipeline/cdc/api/impl/CDCJobAPIImpl.java

```diff
@@ -0,0 +1,195 @@
+        YamlCDCJobConfiguration config = (YamlCDCJobConfiguration) yamlJobConfig;
+        if (null == yamlJobConfig.getJobId()) {
+            config.setJobId(generateJobId(config));
+        }
+        if (Strings.isNullOrEmpty(config.getSourceDatabaseType())) {
+            PipelineDataSourceConfiguration sourceDataSourceConfig = PipelineDataSourceConfigurationFactory.newInstance(config.getDataSourceConfiguration().getType(),
+                    config.getDataSourceConfiguration().getParameter());
+            config.setSourceDatabaseType(sourceDataSourceConfig.getDatabaseType().getType());
+        }
+    }
+    
+    private String generateJobId(final YamlCDCJobConfiguration config) {
+        CDCJobId jobId = new CDCJobId(config.getDatabase(), config.getSubscriptionName());
+        return marshalJobId(jobId);
+    }
+    
+    @Override
+    public PipelineTaskConfiguration buildTaskConfiguration(final PipelineJobConfiguration pipelineJobConfig, final int jobShardingItem, final PipelineProcessConfiguration pipelineProcessConfig) {
+        // TODO to be implement
+        return null;
+    }
+    
+    @Override
+    public PipelineProcessContext buildPipelineProcessContext(final PipelineJobConfiguration pipelineJobConfig) {
+        // TODO to be implement
+        return null;
+    }
+    
+    @Override
+    public PipelineJobConfiguration getJobConfiguration(final String jobId) {
+        return getJobConfiguration(getElasticJobConfigPOJO(jobId));
+    }
+    
+    @Override
+    protected PipelineJobConfiguration getJobConfiguration(final JobConfigurationPOJO jobConfigPOJO) {
+        return new YamlCDCJobConfigurationSwapper().swapToObject(jobConfigPOJO.getJobParameter());
+    }
+    
+    @Override
+    public void persistJobItemProgress(final PipelineJobItemContext jobItemContext) {
+        // TODO to be implement
+    }
+    
+    @Override
+    public PipelineJobItemProgress getJobItemProgress(final String jobId, final int shardingItem) {
+        // TODO to be implement
+        return null;
+    }
+    
+    @Override
+    public void updateJobItemStatus(final String jobId, final int shardingItem, final JobStatus status) {
+        // TODO to be implement
+    }
+    
+    @Override
+    protected String marshalJobIdLeftPart(final PipelineJobId pipelineJobId) {
+        CDCJobId jobId = (CDCJobId) pipelineJobId;
+        String text = Joiner.on('|').join(jobId.getDatabaseName(), jobId.getSubscriptionName());
+        return DigestUtils.md5Hex(text.getBytes(StandardCharsets.UTF_8));
+    }
+    
+    @Override
+    protected PipelineJobInfo getJobInfo(final String jobId) {
+        // TODO to be implement
+        return null;
+    }
+    
+    @Override
+    protected String getJobClassName() {
+        return CDCJob.class.getName();
+    }
+    
+    @Override
+    protected YamlPipelineJobConfiguration swapToYamlJobConfiguration(final PipelineJobConfiguration jobConfig) {
```

Comment:
It's better to put it together with jobConfiguration methods, e.g. getJobConfiguration